### PR TITLE
AP-5220: Remove 'client has home address?' question from CYA

### PR DESCRIPTION
**Pull request by agoldstone93**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-5220)

Remove question from check your answers page if it has not been answered by applicant
Delete address information if applicant has no fixed address
Delete 'care of' information if user answers no to 'care of' question
Fix routing issues when clicking change link for "Where to send client's mail"
Delete `correspondence_address` when changing `correspondence_address_choice` if it is no longer relevant.

## Screenshots

When sending client's mail to "My client's UK home address":
image

When sending client's mail to "Another UK residential address":
image

When sending client's mail to "A UK office address":
image

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by colinbruce — APPROVED** [review]
Looks okay, but the complexity may draw things out during UAT!

File: app/controllers/providers/correspondence_address/care_ofs_controller.rb [hunk 1]
```diff
@@ -10,11 +10,20 @@ def show
       def update
         @form = Addresses::CareOfForm.new(form_params)
 
+        delete_care_of_information if form_params[:care_of].eql?("no")
```
Comment (kmahern) [inline app/controllers/providers/correspondence_address/care_ofs_controller.rb:13]: Does it follow our standard pattern to do this update in the form class?

File: app/controllers/providers/correspondence_address/choices_controller.rb [hunk 2]
```diff
@@ -10,6 +10,12 @@ def show
       def update
         @form = Addresses::ChoiceForm.new(form_params)
 
+        delete_no_fixed_residence if form_params[:correspondence_address_choice].eql?("home")
```
Comment (kmahern) [inline app/controllers/providers/correspondence_address/choices_controller.rb:13]: Does it follow our standard pattern to do this update in the form class?

File: app/controllers/providers/correspondence_address/care_ofs_controller.rb (hunk 1 above)
Comment (agoldstone93) [reply app/controllers/providers/correspondence_address/care_ofs_controller.rb]: Updated in final commit

File: app/controllers/providers/correspondence_address/choices_controller.rb (hunk 2 above)
Comment (agoldstone93) [reply app/controllers/providers/correspondence_address/choices_controller.rb]: Updated in final commit

**Review by kmahern — APPROVED** [review]

**Review by kmahern — APPROVED** [review]
LGTM but agree there is a lot of complexity here and will need a thorough uat.

**Pull request merged**